bench/wtperf/wtperf.c

```diff
@@ -762,6 +762,90 @@ worker(void *arg)
 					value_buf[0] = 'a';
 				if (opts->random_value)
 					randomize_value(thread, value_buf);
+
+				/*
+				 * Distribute the modifications across
+				 * whole document
+				 */
+				if (opts->distribute_modify)
+				{
+					int nentries = 16;
+					size_t iter;
+					size_t total_modify_size;
+					size_t modify_size;
+					size_t modify_offset;
+
+					/* 10% of total size of the record */
+					total_modify_size = strlen(value) / 10;
+
+					/*
+					 * maximum modification size per
+					 * modify
+					 */
+					modify_size = (total_modify_size
+							/ (size_t)nentries);
+
+					/*
+					 * offset location difference between
+					 * modifications
+					 */
+					modify_offset = (strlen(value)
+							/ (size_t)nentries);
+
+					for (iter = (size_t)nentries;
+						iter > 0; iter--)
+						strncpy(
+						&value_buf[
+						    (iter * modify_offset)
+						    - modify_offset],
+						&value_buf[
+						    (iter * modify_offset)
+						    - modify_size],
+						 modify_size);
+
+					/*
+					 * Induce the modify function overhead
+					 * if configured
+					 */
+					if (opts->induce_modify_overhead)
+					{
```

Comment:
Style: the brace (there are a few in this diff) should be at the end of the previous line.

Reply:
Corrected.